src/test/java/unit/GcsBucketReferenced.java

```diff
@@ -0,0 +1,239 @@
+import static unit.GcsBucketControlled.listBucketResourceWithName;
+import static unit.GcsBucketControlled.listOneBucketResourceWithName;
+
+import bio.terra.cli.serialization.userfacing.resources.UFGcsBucket;
+import bio.terra.workspace.model.CloningInstructionsEnum;
+import com.google.cloud.storage.Bucket;
+import harness.TestCommand;
+import harness.baseclasses.SingleWorkspaceUnit;
+import harness.utils.ExternalGCSBuckets;
+import java.io.IOException;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/** Tests for the `terra resources` commands that handle referenced GCS buckets. */
+@Tag("unit")
+public class GcsBucketReferenced extends SingleWorkspaceUnit {
+
+  // external bucket to use for creating GCS bucket references in the workspace
+  private Bucket externalBucket;
+
+  @BeforeEach
+  @Override
+  protected void setupEachTime() throws IOException {
+    super.setupEachTime();
+    externalBucket = ExternalGCSBuckets.createBucket();
+    ExternalGCSBuckets.grantReadAccess(externalBucket, workspaceCreator.email);
+  }
+
+  @AfterEach
+  protected void cleanupEachTime() throws IOException {
+    ExternalGCSBuckets.deleteBucket(externalBucket);
+    externalBucket = null;
+  }
+
+  @Test
+  @DisplayName("list and describe reflect adding a new referenced bucket")
+  void listDescribeReflectAdd() throws IOException {
+    workspaceCreator.login();
+
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getWorkspaceId());
+
+    // `terra resources add-ref gcs-bucket --name=$name --bucket-name=$bucketName --format=json`
+    String name = "listDescribeReflectAdd";
+    UFGcsBucket addedBucket =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFGcsBucket.class,
+            "resources",
+            "add-ref",
+            "gcs-bucket",
+            "--name=" + name,
+            "--bucket-name=" + externalBucket.getName(),
+            "--format=json");
+
+    // check that the name and bucket name match
+    assertEquals(name, addedBucket.name, "add ref output matches name");
+    assertEquals(
+        externalBucket.getName(), addedBucket.bucketName, "add ref output matches bucket name");
+
+    // check that the bucket is in the list
+    UFGcsBucket matchedResource = listOneBucketResourceWithName(name);
+    assertEquals(name, matchedResource.name, "list output matches name");
+    assertEquals(
+        externalBucket.getName(), (matchedResource).bucketName, "list output matches bucket name");
+
+    // `terra resources describe --name=$name --format=json`
+    UFGcsBucket describeResource =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFGcsBucket.class, "resources", "describe", "--name=" + name, "--format=json");
```

Comment:
You could move the `--format=json` into the helper method since you need it every time.

Reply:
Good idea. Done.